runtime/lua/vim/lsp/semantic_tokens.lua

```diff
@@ -637,23 +651,58 @@ end
 --- Only has an effect if the buffer is currently active for semantic token
 --- highlighting (|vim.lsp.semantic_tokens.start()| has been called for it)
 ---
----@param bufnr (nil|number) default: current buffer
+---@param bufnr (number|nil) filter by buffer. All buffers if nil, current
+---       buffer if 0
 function M.force_refresh(bufnr)
   vim.validate({
     bufnr = { bufnr, 'n', true },
   })
 
-  if bufnr == nil or bufnr == 0 then
-    bufnr = api.nvim_get_current_buf()
+  local buffers = bufnr == nil and vim.tbl_keys(STHighlighter.active)
+    or bufnr == 0 and { api.nvim_get_current_buf() }
+    or { bufnr }
+
+  for _, buffer in ipairs(buffers) do
+    local highlighter = STHighlighter.active[buffer]
+    if highlighter then
+      highlighter:reset()
+      highlighter:send_request()
+    end
   end
+end
 
+--- Highlight a semantic token by applying an extmark. This extmark will be
+--- deleted by the semantic token decoration provider on the next `on_win`
+--- update.
+---
```

Comment:
That's not strictly true. It'll be cleared when a new LSP document version gets an updated token response, or a refresh notification is sent by the LSP server (or if the semantic token engine is detached, etc).

Reply:
Yikes, I got this one just _wrong_. I will update.